### ISSUE TITLE
Error deserializing null relationships

### ISSUE THREAD

**Issue by chamini2** [danivek/json-api-serializer#45]
From http://jsonapi.org/format/#crud-updating-to-one-relationships, I understand that a relationship can have a `null` value:

```js
const JSONAPISerializer = require('json-api-serializer');
const Serializer = new JSONAPISerializer({
    convertCase: 'kebab-case',
    unconvertCase: 'snake_case'
});

Serializer.register('requests', {
    id: 'reference_code',
    relationships: {
        coupon: {
            type: 'coupons'
        }
    }
});

Serializer.register('coupons', {
    id: 'code',
    relationships: {
        requests: {
            type: 'requests'
        }
    }
});

const payload = Serializer.deserialize('requests', {
    "data": {
        "id":"50696f63-3f61-41ea-aee2-8512295675c7",
        "type": "activity-requests",
        "relationships": {
            "coupon": { "data": null }
        }
    }
});
```

But this throws:

```
TypeError: Cannot read property 'id' of null
    at Object.keys.forEach
    at Array.forEach (<anonymous>)
    at JSONAPISerializer.deserializeResource
    at JSONAPISerializer.deserialize
```

**Comment by danivek** [comment]
Which version do you use ?

It has been fixed by #44 on last version 1.11.0

**Comment by chamini2** [comment]
Hey, @danivek, I tested this today and it no longer throws, but behaves unexpectedly for me.

When deserialization happens, I expected the value to be `null`, instead, it was not there at all.

I expected this since the specs say:

>The PATCH request MUST include a top-level member named data containing one of:
>
>- a resource identifier object corresponding to the new related resource.
>- null, to remove the relationship.
> 
> ...
>
> And the following request clears the author of the same article:
> ```PATCH /articles/1/relationships/author HTTP/1.1
> Content-Type: application/vnd.api+json
> Accept: application/vnd.api+json
> 
> {
>   "data": null
> }
> ```

**Comment by danivek** [comment]
Fixed in v1.11.1
